src/openrct2/entity/Guest.cpp

```diff
@@ -2305,16 +2308,18 @@ void Guest::SpendMoney(money64& peep_expend_type, money64 amount, ExpenditureTyp
 {
     assert(!(GetGameState().Park.Flags & PARK_FLAGS_NO_MONEY));
 
-    CashInPocket = std::max(0.00_GBP, static_cast<money64>(CashInPocket) - amount);
-    CashSpent += amount;
+    auto modifiedCost = FinanceGetModifiedCost(amount, expenditure);
```

Comment:
I think this is wrong. The modifier should be calculated _after_ the peep gave away its money (but before the player gets the money).